x-pack/plugins/enterprise_search/public/applications/app_search/components/engine/engine_router.tsx

```diff
@@ -69,31 +69,30 @@ export const EngineRouter: React.FC = () => {
     },
   } = useValues(AppLogic);
 
+  const { engineName: engineNameFromUrl } = useParams() as { engineName: string };
   const { engineName, dataLoading, engineNotFound } = useValues(EngineLogic);
   const { setEngineName, initializeEngine, clearEngine } = useActions(EngineLogic);
 
-  const { engineName: engineNameParam } = useParams() as { engineName: string };
-  const engineBreadcrumb = [ENGINES_TITLE, engineNameParam];
-
-  const isEngineInStateStale = () => engineName !== engineNameParam;
-
   useEffect(() => {
-    setEngineName(engineNameParam);
+    setEngineName(engineNameFromUrl);
     initializeEngine();
     return clearEngine;
-  }, [engineNameParam]);
+  }, [engineNameFromUrl]);
 
   if (engineNotFound) {
     setQueuedErrorMessage(
       i18n.translate('xpack.enterpriseSearch.appSearch.engine.notFound', {
-        defaultMessage: "No engine with name '{engineNameParam}' could be found.",
-        values: { engineNameParam },
+        defaultMessage: "No engine with name '{engineName}' could be found.",
+        values: { engineName },
```

Comment:
There isn't really a realistic scenario where `engineName` would get set after `engineNotFound` gets set so I think it's safe to simplify this